convert string felt params, types.Hash, and types.Felt to byte-array backed types.Felt

cfal opened the pull request:
block hashes/transaction hashes (prev. `types.Hash`) are also replaced as they are felt-sized:
  https://github.com/starkware-libs/cairo-lang/blob/c954f154bbab04c3fb27f7598b015a9475fc628e/src/starkware/starknet/definitions/constants.py#L18
  https://github.com/starkware-libs/cairo-lang/blob/c954f154bbab04c3fb27f7598b015a9475fc628e/src/starkware/starknet/definitions/constants.py#L49

cfal committed:
convert external address params to types.Felt

cfal committed:
types/utils.go: temp remove 0 prefix

cfal committed:
replace types.Hash and types.Felt with byte-array backed Felt implementation

cfal committed:
types: remove hash type

archseer commented:
We should be able to remove a bunch of these manual `fmt.Sprintf("0x%s"` calls in various files, e.g.  https://github.com/smartcontractkit/caigo/blob/c8063aace0b1887865c9131f89b37c89becf82f7/rpcv02/mapper.go#L17

cfal committed:
felt.String() returns the shortest possible hex string

cfal committed:
use big.Int, devnet can return larger values than a uint64

json: cannot unmarshal number 25000000000000000000 into Go struct field MintResponse.new_balance of type uint64

cfal committed:
dont prepend zero to hex values
